src/services/navigationBar.ts

```diff
@@ -2,856 +2,551 @@
-
-                            // Handle named bindings in imports e.g.:
-                            //    import * as NS from "mod";
-                            //    import {a, b as B} from "mod";
-                            if (importClause.namedBindings) {
-                                if (importClause.namedBindings.kind === SyntaxKind.NamespaceImport) {
-                                    childNodes.push(importClause.namedBindings);
-                                }
-                                else {
-                                    forEach((<NamedImports>importClause.namedBindings).elements, visit);
-                                }
-                            }
-                        }
-                        break;
-
-                    case SyntaxKind.BindingElement:
-                    case SyntaxKind.VariableDeclaration:
-                        if (isBindingPattern((<VariableDeclaration>node).name)) {
-                            visit((<VariableDeclaration>node).name);
-                            break;
-                        }
-                        // Fall through
-                    case SyntaxKind.ClassDeclaration:
-                    case SyntaxKind.EnumDeclaration:
-                    case SyntaxKind.InterfaceDeclaration:
-                    case SyntaxKind.ModuleDeclaration:
-                    case SyntaxKind.FunctionDeclaration:
-                    case SyntaxKind.ImportEqualsDeclaration:
-                    case SyntaxKind.ImportSpecifier:
-                    case SyntaxKind.ExportSpecifier:
-                    case SyntaxKind.TypeAliasDeclaration:
-                        childNodes.push(node);
-                        break;
-                }
-            }
-
-            // for (let i = 0, n = nodes.length; i < n; i++) {
-            //    let node = nodes[i];
-
-            //    if (node.kind === SyntaxKind.ClassDeclaration ||
-            //        node.kind === SyntaxKind.EnumDeclaration ||
-            //        node.kind === SyntaxKind.InterfaceDeclaration ||
-            //        node.kind === SyntaxKind.ModuleDeclaration ||
-            //        node.kind === SyntaxKind.FunctionDeclaration) {
-
-            //        childNodes.push(node);
-            //    }
-            //    else if (node.kind === SyntaxKind.VariableStatement) {
-            //        childNodes.push.apply(childNodes, (<VariableStatement>node).declarations);
-            //    }
-            // }
-            forEach(nodes, visit);
-            return sortNodes(childNodes);
-        }
-
-        function getTopLevelNodes(node: SourceFile): Node[] {
-            const topLevelNodes: Node[] = [];
-            topLevelNodes.push(node);
-
-            addTopLevelNodes(node.statements, topLevelNodes);
-
-            return topLevelNodes;
-        }
+    export function getNavigationBarItems(sourceFile: SourceFile): NavigationBarItem[] {
+        const root = createNavNode(undefined, sourceFile);

```

Comment:
nit, also add a comment on the undefined parameter target `/*parent:*/ undefined`